File: exercism/kotlin/anagram/src/main/kotlin/Anagram.kt
private fun sortString (s: String): String {
  val letters = s.toCharArray()
  letters.sort()
  return letters.joinToString()
}

class Anagram(word: String) {
  val original = word.toLowerCase()
  val normalized = sortString(original)

  fun match(others: List<String>): List<String> {
    return others.filter {
      val word = it.toLowerCase()
      word != original && sortString(word) == normalized
    }
  }
}

File: exercism/haskell/anagram/src/Anagram.hs
module Anagram where
import Data.List (sort)
import Data.Char (toLower)

anagramsFor :: String -> [String] -> [String]
anagramsFor word = sortWord `seq` filter (isAnagram . low)
  where
    isAnagram t = (unequal t) && (sameLetters t)
    unequal = (lowWord /=)
    sameLetters = (sortWord ==) . sort
    low = map toLower
    lowWord = low word
    sortWord = sort lowWord
